src/ucp/stream/stream_recv.c

```diff
@@ -304,7 +304,7 @@ ucp_stream_am_handler(void *am_arg, void *am_data, size_t am_length,
 
     ep_stream = ep->ext.stream;
 
-    if (ep->flags & UCP_EP_FLAG_STREAM_IS_QUEUED) {
+    if (ucp_stream_ep_is_queued(ep)) {
```

Comment:
need to do ucs_queue_push(&ep_stream->data, &rdesc->stream_queue) before this check to avoid code duplication